### PR TITLE
Make dropdown menu text not editable and fix dropdown related issues

### PR DESCRIPTION
**Pull request by praveennadarajah**
## Purpose
Fixing combo box scroll issue
Resolves: https://github.com/wso2/product-ei/issues/3975

### CODE REVIEW

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/MessageProcessorPropertiesEditionPartForm.java [hunk 1]
```diff
@@ -787,11 +796,18 @@ public void keyPressed(KeyEvent e) {
 	 */
 	protected Composite createProcessorStateEMFComboViewer(FormToolkit widgetFactory, Composite parent) {
 		createDescription(parent, EsbViewsRepository.MessageProcessor.Properties.processorState, EsbMessages.MessageProcessorPropertiesEditionPart_ProcessorStateLabel);
-		processorState = new EMFComboViewer(parent, SWT.SCROLL_LOCK);
+		processorState = new EMFComboViewer(parent);
 		processorState.setContentProvider(new ArrayContentProvider());
 		processorState.setLabelProvider(new AdapterFactoryLabelProvider(EEFRuntimePlugin.getDefault().getAdapterFactory()));
 		GridData processorStateData = new GridData(GridData.FILL_HORIZONTAL);
 		processorState.getCombo().setLayoutData(processorStateData);
+                processorType.getCombo().addListener(SWT.MouseVerticalWheel, new Listener() {
```
Comment (rosensilva) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/MessageProcessorPropertiesEditionPartForm.java]: processorType should be processorState

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/MessageStorePropertiesEditionPartForm.java [hunk 2]
```diff
@@ -1782,11 +1798,18 @@ public void keyPressed(KeyEvent e) {
 	 */
 	protected Composite createJdbcConnectionInformationEMFComboViewer(FormToolkit widgetFactory, Composite parent) {
 		createDescription(parent, EsbViewsRepository.MessageStore.Properties.jdbcConnectionInformation, EsbMessages.MessageStorePropertiesEditionPart_JdbcConnectionInformationLabel);
-		jdbcConnectionInformation = new EMFComboViewer(parent, SWT.SCROLL_LOCK);
+		jdbcConnectionInformation = new EMFComboViewer(parent);
 		jdbcConnectionInformation.setContentProvider(new ArrayContentProvider());
 		jdbcConnectionInformation.setLabelProvider(new AdapterFactoryLabelProvider(EEFRuntimePlugin.getDefault().getAdapterFactory()));
 		GridData jdbcConnectionInformationData = new GridData(GridData.FILL_HORIZONTAL);
 		jdbcConnectionInformation.getCombo().setLayoutData(jdbcConnectionInformationData);
+                jmsSpecVersion.getCombo().addListener(SWT.MouseVerticalWheel, new Listener() {
```
Comment (rosensilva) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/MessageStorePropertiesEditionPartForm.java]: jmsSpecVersion should be jdbcConnectionInformation

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/ProxyServicePropertiesEditionPartForm.java [hunk 3]
```diff
@@ -1140,11 +1156,18 @@ protected Composite createInSequenceGroup(FormToolkit widgetFactory, final Compo
      */
 	protected Composite createInSequenceTypeEMFComboViewer(FormToolkit widgetFactory, Composite parent) {
 		Control inSeqTypeLabel = createDescription(parent, EsbViewsRepository.ProxyService.InSequence.inSequenceType, EsbMessages.ProxyServicePropertiesEditionPart_InSequenceTypeLabel);
-		inSequenceType = new EMFComboViewer(parent, SWT.SCROLL_LOCK);
+		inSequenceType = new EMFComboViewer(parent);
 		inSequenceType.setContentProvider(new ArrayContentProvider());
 		inSequenceType.setLabelProvider(new AdapterFactoryLabelProvider(EEFRuntimePlugin.getDefault().getAdapterFactory()));
 		GridData inSequenceTypeData = new GridData(GridData.FILL_HORIZONTAL);
 		inSequenceType.getCombo().setLayoutData(inSequenceTypeData);
+                faultSequenceType.getCombo().addListener(SWT.MouseVerticalWheel, new Listener() {
```
Comment (rosensilva) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/ProxyServicePropertiesEditionPartForm.java]: faultSequenceType should be inSequenceType

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/RMSequenceMediatorPropertiesEditionPartForm.java [hunk 4]
```diff
@@ -362,11 +371,18 @@ public void selectionChanged(SelectionChangedEvent event) {
 	 */
 	protected Composite createSequenceTypeEMFComboViewer(FormToolkit widgetFactory, Composite parent) {
 		createDescription(parent, EsbViewsRepository.RMSequenceMediator.Properties.sequenceType, EsbMessages.RMSequenceMediatorPropertiesEditionPart_SequenceTypeLabel);
-		sequenceType = new EMFComboViewer(parent, SWT.SCROLL_LOCK);
+		sequenceType = new EMFComboViewer(parent);
 		sequenceType.setContentProvider(new ArrayContentProvider());
 		sequenceType.setLabelProvider(new AdapterFactoryLabelProvider(EEFRuntimePlugin.getDefault().getAdapterFactory()));
 		GridData sequenceTypeData = new GridData(GridData.FILL_HORIZONTAL);
 		sequenceType.getCombo().setLayoutData(sequenceTypeData);
+                rmSpecVersion.getCombo().addListener(SWT.MouseVerticalWheel, new Listener() {
```
Comment (rosensilva) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/RMSequenceMediatorPropertiesEditionPartForm.java]: sequenceType should be rmSpecVersion

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/MessageStorePropertiesEditionPartForm.java (hunk 2 above)
Comment (praveennadarajah) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/MessageStorePropertiesEditionPartForm.java]: jmsSpecVersion should be jdbcConnectionInformation

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/SynapseAPIPropertiesEditionPartForm.java [hunk 5]
```diff
@@ -703,11 +705,18 @@ public void widgetSelected(SelectionEvent e) {
...
 		versionType.setLabelProvider(new AdapterFactoryLabelProvider(EEFRuntimePlugin.getDefault().getAdapterFactory()));
 		GridData versionTypeData = new GridData(GridData.FILL_HORIZONTAL);
 		versionType.getCombo().setLayoutData(versionTypeData);
+               versionType.getCombo().addListener(SWT.MouseVerticalWheel, new Listener() {
+
+                   @Override
+                   public void handleEvent(Event arg0) {
+                       arg0.doit = false;
+                   }
+               });
```
Comment (rosensilva) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/SynapseAPIPropertiesEditionPartForm.java]: Formatting issue

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/ProxyServicePropertiesEditionPartForm.java (hunk 3 above)
Comment (praveennadarajah) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/ProxyServicePropertiesEditionPartForm.java]: faultSequenceType should be inSequenceType

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/RMSequenceMediatorPropertiesEditionPartForm.java (hunk 4 above)
Comment (praveennadarajah) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/RMSequenceMediatorPropertiesEditionPartForm.java]: rmSpecVersion should be sequenceType

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/SynapseAPIPropertiesEditionPartForm.java [hunk 6]
```diff
@@ -703,11 +705,18 @@ public void widgetSelected(SelectionEvent e) {
      */	
 	protected Composite createVersionTypeEMFComboViewer(FormToolkit widgetFactory, Composite parent) {
 		Control versionTypeLabel = createDescription(parent, EsbViewsRepository.SynapseAPI.Handler.Properties.versionType, EsbMessages.SynapseAPIPropertiesEditionPart_VersionTypeLabel);
-		versionType = new EMFComboViewer(parent, SWT.SCROLL_LOCK);
+		versionType = new EMFComboViewer(parent);
 		versionType.setContentProvider(new ArrayContentProvider());
 		versionType.setLabelProvider(new AdapterFactoryLabelProvider(EEFRuntimePlugin.getDefault().getAdapterFactory()));
 		GridData versionTypeData = new GridData(GridData.FILL_HORIZONTAL);
 		versionType.getCombo().setLayoutData(versionTypeData);
+               versionType.getCombo().addListener(SWT.MouseVerticalWheel, new Listener() {
```
Comment (praveennadarajah) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/SynapseAPIPropertiesEditionPartForm.java]: Code format issue

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/impl/DBReportMediatorPropertiesEditionPartImpl.java [hunk 7]
```diff
@@ -1265,11 +1316,18 @@ public void selectionChanged(SelectionChangedEvent event) {
 	 */
 	protected Composite createPropertyTestwhileidleEMFComboViewer(Composite parent) {
 		createDescription(parent, EsbViewsRepository.DBReportMediator.Properties.propertyTestwhileidle, EsbMessages.DBReportMediatorPropertiesEditionPart_PropertyTestwhileidleLabel);
-		propertyTestwhileidle = new EMFComboViewer(parent, SWT.SCROLL_LOCK);
+		propertyTestwhileidle = new EMFComboViewer(parent);
 		propertyTestwhileidle.setContentProvider(new ArrayContentProvider());
 		propertyTestwhileidle.setLabelProvider(new AdapterFactoryLabelProvider(EEFRuntimePlugin.getDefault().getAdapterFactory()));
 		GridData propertyTestwhileidleData = new GridData(GridData.FILL_HORIZONTAL);
 		propertyTestwhileidle.getCombo().setLayoutData(propertyTestwhileidleData);
+                propertyTestonborrow.getCombo().addListener(SWT.MouseVerticalWheel, new Listener() {
```
Comment (praveennadarajah) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/impl/DBReportMediatorPropertiesEditionPartImpl.java]: propertyTestonborrow should be propertyTestwhileidle

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/impl/InboundEndpointPropertiesEditionPartImpl.java [hunk 8]
```diff
@@ -6039,11 +6139,18 @@ public void keyPressed(KeyEvent e) {
 	 */
 	protected Composite createTopicFilterFromEMFComboViewer(Composite parent) {
 		createDescription(parent, EsbViewsRepository.InboundEndpoint.Properties.topicFilterFrom, EsbMessages.InboundEndpointPropertiesEditionPart_TopicFilterFromLabel);
-		topicFilterFrom = new EMFComboViewer(parent, SWT.SCROLL_LOCK);
+		topicFilterFrom = new EMFComboViewer(parent);
 		topicFilterFrom.setContentProvider(new ArrayContentProvider());
 		topicFilterFrom.setLabelProvider(new AdapterFactoryLabelProvider(EEFRuntimePlugin.getDefault().getAdapterFactory()));
 		GridData topicFilterFromData = new GridData(GridData.FILL_HORIZONTAL);
 		topicFilterFrom.getCombo().setLayoutData(topicFilterFromData);
+                topicsOrTopicFilter.getCombo().addListener(SWT.MouseVerticalWheel, new Listener() {
```
Comment (praveennadarajah) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/impl/InboundEndpointPropertiesEditionPartImpl.java]: topicsOrTopicFilter should be topicFilterFrom

**Pull request merged**